cli/src/test/kotlin/com/bazel_diff/e2e/E2ETest.kt

```diff
@@ -179,6 +210,56 @@ class E2ETest {
         assertThat(actual).isEqualTo(expected)
     }
 
+//     @Ignore("foo")
+    @Test
+    fun testTargetDistanceMetrics() {
+        val workspace = copyTestWorkspace("distance_metrics")
+        println("Testing in workspace: $workspace")
+
+        val outputDir = temp.newFolder()
+        val from = File(outputDir, "starting_hashes.json")
+        val to = File(outputDir, "final_hashes.json")
+        val depsFile = File(outputDir, "depEdges.json")
+        val impactedTargetsOutput = File(outputDir, "impacted_targets.txt")
+
+        val cli = CommandLine(BazelDiff())
+
+        cli.execute("generate-hashes", "--includeTargetType", "-w", workspace.absolutePath, "-b", "bazel", from.absolutePath)
+        // Modify the workspace
+        File(workspace, "A/one.sh").appendText("foo")
+        cli.execute("generate-hashes", "--includeTargetType", "-w", workspace.absolutePath, "-d", depsFile.absolutePath, "-b", "bazel", to.absolutePath)
+
+        //Impacted targets
+        cli.execute(
+                "get-impacted-targets", 
+                "-sh", from.absolutePath, 
+                "-fh", to.absolutePath, 
+                "-d", depsFile.absolutePath,
+                "-tt", "Rule,GeneratedFile",
+                "-o", impactedTargetsOutput.absolutePath, 
+        )
+
+        // When computing target distances, the output format is json. Read the files and assert the sorted contents.
+        val gson = Gson()
+        val shape = object : TypeToken<List<Map<String, Any>>>() {}.type
+        val actual = gson.fromJson<List<Map<String, Any>>>(impactedTargetsOutput.readText(), shape).sortedBy { it["label"] as String }
+        val expected: List<Map<String, Any>> = listOf(
+                mapOf("label" to "//A:one", "targetDistance" to 0.0, "packageDistance" to 0.0),
+                mapOf("label" to "//A:gen_two", "targetDistance" to 1.0, "packageDistance" to 0.0),
+                mapOf("label" to "//A:two.sh", "targetDistance" to 2.0, "packageDistance" to 0.0),
+                mapOf("label" to "//A:two", "targetDistance" to 3.0, "packageDistance" to 0.0),
```

Comment:
NOTE: GeneratedFiles contribute to targetDistance. I could see the argument for not doing that and instead having the code rewire the dependencies so that `//A:two` directly depends on `//A:gen_two` and the GeneratedFile doesn't contribute to targetDistance.

Reply:
Sounds good